server/src/tree-sitter/analyzer.ts

```diff
@@ -701,6 +702,64 @@ export default class Analyzer {
       }
     }
   }
+
+  public processRecipeScanResults (scanResults: string, uri: any): RequestResult['ProcessRecipeScanResults'] {
+    let uriString
+    if (typeof uri === 'string') {
+      uriString = uri
+    }
+
+    if (uri.fsPath !== undefined) { // vscode.Uri
+      uriString = uri.fsPath
+    }
+
+    if (typeof uriString !== 'string') {
+      logger.debug('[ProcessRecipeScanResults] Cannot obtain the file uri, abort processing scan results')
+      return undefined
+    }
+
+    if (!uriString.startsWith('file://')) {
+      uriString = 'file://' + uriString
+    }
+
+    if (this.uriToAnalyzedDocument[uriString] === undefined) {
+      logger.debug(`[ProcessRecipeScanResults] Analyzed document for ${uriString} not found, abort processing scan results`)
+      return undefined
+    }
+
+    const lines = scanResults.split('\n')
+    const index = lines.findIndex((line) => line.includes('INCLUDE HISTORY'))
```

Comment:
This function is doing several things. You could consider splitting it into subfunctions:
 1. format and validate the URI
 2. split the INCLUDE HISTORY (and probably parse it later)
 3. generate and contribute symbols scan

Reply:
Can you review this part in https://github.com/yoctoproject/vscode-bitbake/pull/73 as I made many changes to this function? If it is only refactoring then it is easier for me because I won't need to solve the conflicts.